### PR TITLE
Fix check for world_discoverable

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made? 🤔

Fixes https://github.com/sul-dlss/common-accessioning/issues/840 (at least part of)

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: app/services/publish/metadata_transfer_service.rb [hunk 1]
```diff
@@ -47,7 +47,9 @@ def unpublish
     end
 
     def world_discoverable?
-      ['world', 'citation-only'].include? cocina_object.access.access
+      return false if ['dark', 'none'].include? cocina_object.access.access
```
Comment (mjgiarlo) [inline app/services/publish/metadata_transfer_service.rb:50]: AFAIK, there are only five valid values for `access.access` across our models and those are world, stanford, location-based, citation-only, and dark. So I am wondering if `'none'` can be removed from this check. Cc: @andrewjbtw

Comment (aaron-collier) [reply app/services/publish/metadata_transfer_service.rb:50]: Oh, that's likely @mjgiarlo - I'd have to hunt it down, but I've seen `none`... somewhere... and thought to include it. But if it's unnecessary, happy to remove.

Comment (jcoyne) [reply app/services/publish/metadata_transfer_service.rb:50]: Since include? is a boolean, can you remove the `if` statement?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/services/publish/metadata_transfer_service.rb (hunk 1 above)
Comment (jcoyne) [reply app/services/publish/metadata_transfer_service.rb:50]: "none" is possible for download, but not for access. See https://github.com/sul-dlss/cocina-models/blob/main/openapi.yml

Comment (aaron-collier) [reply app/services/publish/metadata_transfer_service.rb]: Ah, yeah, that's it.. Thanks @jcoyne

File: app/services/publish/metadata_transfer_service.rb [hunk 2]
```diff
@@ -47,7 +47,7 @@ def unpublish
     end
 
     def world_discoverable?
```
Comment (jcoyne) [inline app/services/publish/metadata_transfer_service.rb:49]: Can we change the name please? "world_discoverable?" is misleading. Now "discoverable?" seems like a better name.

File: app/services/publish/metadata_transfer_service.rb [hunk 3]
```diff
@@ -47,7 +47,7 @@ def unpublish
...
-      ['world', 'citation-only'].include? cocina_object.access.access
+      %w[world stanford location-based citation-only].include? cocina_object.access.access
```
Comment (jcoyne) [inline app/services/publish/metadata_transfer_service.rb:50]: Would it be easier to say `!= 'dark'`?

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Review by jcoyne — APPROVED** [review]

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**